### PR TITLE
feat: larger and configurable receptive field for cursor in 3D

### LINKED ISSUE #225: Fuzzy 3D Object Selection
For object picking, Neuroglancer only checks the pixel at the current mouse position. Which makes it hard for the user to select thin structures in the 3D view.
Changing the behavior to something smarter, like an incremental search from the mouse position with increasing radii would make this task easier/faster.

This also affects google/neuroglancer, so it would be nice if we could send the PR there, too.

### PR DESCRIPTION
**Pull request by william-silversmith**
Resolves #225

The receptive field is square, so the diagonals are more extensive. Would going circular be beneficial?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,7 +77,7 @@
     "npm": "^5.6.0",
     "pako": "^1.0.6",
     "raw-loader": "^0.5.1",
-    "sockjs-client": "^1.1.1",
+    "sockjs-client": "^1.3.0",
```
[the remark this answers is not in the document]
Comment (william-silversmith) [reply package.json]: I ran `npm audit fix` which is the source of this chang.e I think there might have been a security issue.

File: src/neuroglancer/perspective_view/panel.ts [hunk 2]
```diff
@@ -349,11 +356,36 @@ export class PerspectivePanel extends RenderedDataPanel {
...
+        break;
+      }
+    }
+
+    let glWindowZ = 1.0 - zDatum;
     if (glWindowZ === 1.0) {
       return false;
     }
+
+    glWindowX += (rfindex % field_width) - (field_width >> 1);
```
[the remark this answers is not in the document]
Comment (william-silversmith) [reply src/neuroglancer/perspective_view/panel.ts]: Fixed.

File: src/neuroglancer/webgl/offscreen.ts [hunk 3]
```diff
@@ -217,35 +217,34 @@ export class FramebufferConfiguration<ColorBuffer extends TextureBuffer|Renderbu
...
-    let cutout = new Uint8Array(width * height * 4);
-
-    let left = Math.max(0, glWindowX - (width >> 1))
-    let bottom = Math.max(0, glWindowY - (height >> 1))
+    
+    if (!buffer) {
+      buffer = new Uint8Array(width * height * 4);
+    }
+    
+    // Appearently WebGL supports reading pixels off the
```
Comment (nkemnitz) [inline src/neuroglancer/webgl/offscreen.ts]: Supports, but doesn't necessarily zero them out. From the [specification](https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glReadPixels.xml):
> Values for pixels that lie outside the window connected to the current GL context are undefined.

Comment (william-silversmith) [reply src/neuroglancer/webgl/offscreen.ts]: Thanks for the link, I couldn't find that spec before.

Comment (william-silversmith) [reply src/neuroglancer/webgl/offscreen.ts]: Hmm, this problem is probably true in the positive direction as well.

**Pull request merged**